File: frontend/src/modules/analytics/services/analyticsService.js
import axiosInstance from "../../../api/axiosInstance";

function normalizeDashboardData(responseData) {
  const raw = responseData?.data ?? responseData ?? {};

  return {
    totalUsers: Number(raw.totalUsers ?? 0),
    totalCases: Number(raw.totalCases ?? 0),
    totalSubmissions: Number(raw.totalSubmissions ?? 0),
    activeCases: Number(raw.activeCases ?? 0),
    pendingReviews: Number(raw.pendingReviews ?? 0),
    activeFaculty: Number(raw.activeFaculty ?? 0),
  };
}

const analyticsService = {
  async getDashboardStats() {
    const response = await axiosInstance.get("/admin/analytics/dashboard");
    return normalizeDashboardData(response.data);
  },

  async getUserAnalytics() {
    const response = await axiosInstance.get("/admin/analytics/users");
    return response.data;
  },

  async getCaseAnalytics() {
    const response = await axiosInstance.get("/admin/analytics/cases");
    return response.data;
  },

  async getSubmissionAnalytics() {
    const response = await axiosInstance.get("/admin/analytics/submissions");
    return response.data;
  },

  async getCoAttainmentSummary() {
    const response = await axiosInstance.get("/admin/analytics/co-attainment-summary");
    return response.data;
  },

  async getTopCases() {
    const response = await axiosInstance.get("/admin/analytics/top-cases");
    return response.data;
  },

  async getFacultyPerformance() {
    const response = await axiosInstance.get("/admin/faculty-performance");
    return Array.isArray(response.data) ? response.data : [];
  },

  async getOverallStats() {
    const response = await axiosInstance.get("/admin/overall-stats");
    return response.data ?? {};
  },

  async getFacultyStudentsBreakdown(facultyId) {
    if (facultyId === null || facultyId === undefined || facultyId === "") {
      throw new Error("Faculty ID is required for student breakdown.");
    }
    const response = await axiosInstance.get(`/admin/faculty-performance/${facultyId}/students`);
    return response.data ?? null;
  },
};

export default analyticsService;

File: backend/src/main/java/com/icr/backend/service/AnalyticsService.java
package com.icr.backend.service;

import com.icr.backend.dto.DashboardStatsResponse;
import com.icr.backend.dto.FacultyPerformanceDTO;
import com.icr.backend.dto.FacultyStudentsBreakdownDTO;
import com.icr.backend.dto.OverallStatsDTO;
import com.icr.backend.dto.CoAttainmentSummaryDTO;
import com.icr.backend.dto.TopCaseAnalyticsDTO;

import java.util.List;
import java.util.Map;

public interface AnalyticsService {

    DashboardStatsResponse getDashboardStats();

    Map<String, Long> getUserAnalytics();

    Map<String, Long> getCaseAnalytics();

    Map<String, Long> getSubmissionAnalytics();

    List<CoAttainmentSummaryDTO> getCoAttainmentSummary();

    List<TopCaseAnalyticsDTO> getTopCases();

    List<FacultyPerformanceDTO> getFacultyPerformance();

    FacultyStudentsBreakdownDTO getFacultyStudentsBreakdown(Long facultyId);

    OverallStatsDTO getOverallStats();
}
